pyangbind/lib/serialise.py

```diff
@@ -444,6 +444,112 @@ def generate_ietf_tree(obj, parent_namespace=None, flt=False, with_defaults=None
     return generate_ietf_tree
 
 
+class pybindIETFXMLDecoder(object):
+    """
+    IETF XML decoder for pybind object tree deserialisation.
+    Use the `decode()` method to return an pyangbind representation of the yang object.
+    """
+
+    @classmethod
+    def decode(cls, xml, bindings, module_name):
+        # using a custom parser to strip comments (so we don't handle them later)
+        parser = objectify.makeparser(remove_comments=True, remove_blank_text=True)
+        doc = objectify.fromstring(xml, parser=parser)
+        return cls.load_xml(doc, bindings, module_name)
+
+    @staticmethod
+    def load_xml(d, parent, yang_base, obj=None, path_helper=None, extmethods=None):
+        """low-level XML deserialisation function, based on pybindJSONDecoder.load_ietf_json()"""
+        if obj is None:
+            # we need to find the class to create, as one has not been supplied.
+            base_mod_cls = getattr(parent, safe_name(yang_base))
+            tmp = base_mod_cls(path_helper=False)
+
+            if path_helper is not None:
+                # check that this path doesn't already exist in the
+                # tree, otherwise we create a duplicate.
+                existing_objs = path_helper.get(tmp._path())
+                if len(existing_objs) == 0:
+                    obj = base_mod_cls(path_helper=path_helper, extmethods=extmethods)
+                elif len(existing_objs) == 1:
+                    obj = existing_objs[0]
+                else:
+                    raise pybindJSONUpdateError("update was attempted to a node that " + "was not unique")
```

Comment:
Maybe use a different exception name -- `pybindJSONUpdateError` might be confusing given that this is XML. Perhaps we just need a common decode error?

Reply:
I've renamed the exception. Naming is hard. Let me know if you have a better suggestion.